common/utils/src/main/resources/error/error-conditions.json

```diff
@@ -2798,6 +2798,11 @@
     ],
     "sqlState" : "42601"
   },
+  "INVALID_STATE_STORE_PROVIDER" : {
+    "message" : [
+      "The given State Store Provider <inputClass> does not extend org.apache.spark.sql.execution.streaming.state.StateStoreProvider."
```

Comment:
I believe u also need to define some `sqlState` ?

Reply:
Yep. Done. I choose this state:
```
"42K06": {
        "description": "Invalid type for options",
        "origin": "Spark",
        "standard": "N",
        "usedBy": ["Spark"]
    }
```